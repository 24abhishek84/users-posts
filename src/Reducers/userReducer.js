export const initialState = {
    mainUserList: [],
    userList: [],
    currentRecord: null,
    userPosts: [],
    searchText: ''
};

const common = (state = initialState, action = {}) => {
  const {mainUserList, searchText} = state;
  switch (action.type) {
    case 'SAVE_USER_LIST':
      return {...state, mainUserList: [...action.payload], userList: [...action.payload]};
    case 'FILTER_USER_LIST':
      console.log('action.payload', action.payload);
      console.log('searchText', state.searchText);
      const search = action.payload ? action.payload.toLowerCase() : searchText;
      const filteredUserList = search ? mainUserList.filter(x => x.name.toLowerCase().includes(search)) : [...mainUserList];
      return {...state, userList: [...filteredUserList], searchText: action.payload};
    case 'SET_CURRENT_RECORD':
      console.log('action.payload', action.payload);
      return {...state, currentRecord: action.payload?.id};
    case 'SAVE_USER_POSTS':
      return {...state, userPosts: [...action.payload]};
      default: return state;
  }
};

export default common;