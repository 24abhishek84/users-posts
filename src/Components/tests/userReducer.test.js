import common from '../../Reducers/userReducer';
import {initialState} from '../../Reducers/userReducer';

describe('users reducer', () => {
  it('should return the initial state', () => {
    expect(common(undefined, {})).toMatchSnapshot()
  })

  const record = {
    id: 2,
    name: 'Sam Dun',
    email: 'sam@ddun.com',
    company: 'Sam Company'
  };
  const payload = [
    {
      id: 1,
      name: 'John Doe',
      email: 'john@document.com',
      company: 'Doe Company'
    },
    {...record}
  ];

  it('should handle SAVE_USER_LIST', () => {
    expect(
      common(initialState,
      {
        type: 'SAVE_USER_LIST',
        payload
      })
    ).toMatchSnapshot()
  });

  it('should handle SET_CURRENT_RECORD', () => {
    expect(
      common(initialState,
      {
        type: 'SET_CURRENT_RECORD',
        payload: {...record}
      })
    ).toMatchSnapshot()
  })
})

