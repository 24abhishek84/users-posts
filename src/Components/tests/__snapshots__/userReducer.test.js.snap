// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users reducer should handle SAVE_USER_LIST 1`] = `
Object {
  "currentRecord": null,
  "mainUserList": Array [
    Object {
      "company": "Doe Company",
      "email": "john@document.com",
      "id": 1,
      "name": "John Doe",
    },
    Object {
      "company": "Sam Company",
      "email": "sam@ddun.com",
      "id": 2,
      "name": "Sam Dun",
    },
  ],
  "searchText": "",
  "userList": Array [
    Object {
      "company": "Doe Company",
      "email": "john@document.com",
      "id": 1,
      "name": "John Doe",
    },
    Object {
      "company": "Sam Company",
      "email": "sam@ddun.com",
      "id": 2,
      "name": "Sam Dun",
    },
  ],
  "userPosts": Array [],
}
`;

exports[`users reducer should handle SET_CURRENT_RECORD 1`] = `
Object {
  "currentRecord": 2,
  "mainUserList": Array [],
  "searchText": "",
  "userList": Array [],
  "userPosts": Array [],
}
`;

exports[`users reducer should handle TODO_REQUEST 1`] = `
Object {
  "currentRecord": null,
  "mainUserList": Array [
    Object {
      "company": "Doe Company",
      "email": "john@document.com",
      "id": 1,
      "name": "John Doe",
    },
  ],
  "searchText": "",
  "userList": Array [
    Object {
      "company": "Doe Company",
      "email": "john@document.com",
      "id": 1,
      "name": "John Doe",
    },
  ],
  "userPosts": Array [],
}
`;

exports[`users reducer should return the initial state 1`] = `
Object {
  "currentRecord": null,
  "mainUserList": Array [],
  "searchText": "",
  "userList": Array [],
  "userPosts": Array [],
}
`;
